docs/schema/ilab-config-version.md

```diff
@@ -0,0 +1,27 @@
+# `ilab` Config.yaml Versioning
+
+## Problem Statement
+
+Currently the `ilab` CLI's configuration file, `config.yaml`, does not contain a version.
+
+The config file in `ilab` version `0.17` cannot be used in future versions of `ilab` due to new mandatory fields and changes to existing fields.
+
+As the configuration file grows, changes, and becomes more complex, versioning is necessary in order to support forwards and backwards compatibility.
+
+## Proposal
+
+The `ilab` configuration file has a new `version` field at the top level of the config.
+
+The value of `version` should start at `1.0`.
```

Comment:
```suggestion
The value of `version` should start at `1`.
```

I would just use a simple integer unless you plan to handle semantic versioning. Like where the cli handles all 1.x variants but not 2.x.